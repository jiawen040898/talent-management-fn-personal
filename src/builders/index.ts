export * from './base-query.builder';
export * from './employee-task-query.builder';
export * from './feedback-cycle-reviewee-query.builder';
export * from './feedback-cycle-reviewer-query.builder';
export * from './learning-goal-query.builder';
