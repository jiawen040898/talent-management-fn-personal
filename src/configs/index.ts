export * from './aws.config';
export * from './gcp.config';
export * from './recommendation.config';
