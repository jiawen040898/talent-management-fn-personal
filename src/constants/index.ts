export * from './action-source.enum';
export * from './big-queries.enum';
export * from './cache-store.enum';
export * from './company-module.enum';
export * from './company-status.enum';
export * from './due-type.enum';
export * from './employee-score.enum';
export * from './employee-task.enum';
export * from './event-type.enum';
export * from './feedback-recommendation.enum';
export * from './fit-score-recipe.enum';
export * from './person-score.enum';
export * from './psychology-api-errors.enum';
export * from './questionnaire-framework-type.enum';
export * from './sqs-event-type.enum';
export * from './unleash-errors.enum';
