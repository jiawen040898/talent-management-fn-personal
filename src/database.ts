import { secretService } from '@pulsifi/fn';
import { DataSource } from 'typeorm';

import { ENTITIES } from './models';

type DbSecret = {
    username: string;
    engine: string;
    port: number;
    host: string;
    password: string;
    schema: string;
    dbname: string;
    dbInstanceIdentifier: string;
};

let dataSource: DataSource;

export const getDataSource = async (): Promise<DataSource> => {
    if (dataSource?.isInitialized) {
        return dataSource;
    }

    const dbSecret = await secretService.getSecret<DbSecret>(
        process.env.SM_NAME as string,
    );

    dataSource = await new DataSource({
        type: 'postgres',
        host: dbSecret.host,
        port: dbSecret.port,
        username: dbSecret.username,
        password: dbSecret.password,
        database: dbSecret.dbname,
        schema: dbSecret.schema,
        entities: [...ENTITIES],
        maxQueryExecutionTime: 1000,
        synchronize: false,
        migrationsRun: false,
        migrations: [],
        extra: {
            connectionTimeoutMillis: 3000,
            idleTimeoutMillis: 1000,
            max: 1,
        },
        logging: ['error'],
    }).initialize();

    return dataSource;
};
