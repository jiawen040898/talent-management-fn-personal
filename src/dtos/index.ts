export * from './company.dto';
export * from './cron-schedule-message.dto';
export * from './employee.dto';
export * from './employee-account-created-event.dto';
export * from './employee-assessment-score-event.dto';
export * from './employee-fit-score-event.dto';
export * from './feedback-cycle.dto';
export * from './fit-score-recipe.dto';
export * from './manager-subordinate-aggregated-report-data.dto';
