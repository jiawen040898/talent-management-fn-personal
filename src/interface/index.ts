export * from '../dtos/fit-score-recipe.dto';
export * from './cognitive-score.interface';
export * from './employee-score.interface';
export * from './event-model.interface';
export * from './feedback-cycle.interface';
export * from './feedback-recommendation.interface';
export * from './fit-score.interface';
export * from './personality-score.interface';
export * from './program.interface';
export * from './redis.interface';
