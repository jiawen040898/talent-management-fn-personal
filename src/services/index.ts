export * from './cognitive-score.service';
export * from './company.service';
export * from './domain-event-router.service';
export * from './employee-daily-reminder.service';
export * from './employee-management.service';
export * from './employee-progress-report.service';
export * from './employee-score.service';
export * from './feedback-cycle-action.service';
export * from './feedback-cycle-dashboard-provision.service';
export * from './feedback-cycle-reminder.service';
export * from './feedback-cycle-reviewee-recommendation.service';
export * from './fit-score.service';
export * from './personality-score.service';
export * from './psychology.service';
export * from './recommendation.service';
