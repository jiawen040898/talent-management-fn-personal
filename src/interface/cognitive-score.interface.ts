import { EmployeeScoreType } from '../constants';
import { PartialScoreOutput } from './employee-score.interface';

export class CognitiveOutcomeDto {
    domain_alias?: string;
    domain_score!: number;
    domain_percentile!: number | null;
    ingredient_weightage?: number | null;
}

export type CognitiveResult = {
    [K in keyof CognitiveOutcomeDto]?: CognitiveOutcomeDto[K] | null;
};

export interface CognitiveScoreOutcome {
    score_type: EmployeeScoreType;
    outcome: CognitiveOutcomeDto;
}

export interface CognitiveScoreOutcomePayload {
    cognitive_result: CognitiveOutcomeDto;
}

export interface CognitivePartialScoreOutput extends PartialScoreOutput {
    score_outcome: CognitiveScoreOutcomePayload;
    score: number;
}
