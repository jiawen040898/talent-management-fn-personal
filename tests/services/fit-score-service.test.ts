import { IngredientFramework } from '@pulsifi/fn';
import { ParticipantPartialScoreInput } from 'src/interface';

import { FitScoreService } from '../../src/services';
import { cultureFitRecipeTestData } from './fixtures/culture-fit-recipe.test-data';
import { employeeScoreTestData } from './fixtures/employee-score-service.test-data';
import { fitScoreTestData } from './fixtures/fit-score-service.test-data';
import { roleFitRecipeTestData } from './fixtures/role-fit-recipe.test-data';

describe('FitScoreService', () => {
    describe('processCultureFitDomainScore', () => {
        it('should return expected domain outcome', () => {
            // Act
            const result = FitScoreService.processCultureFitDomainScore(
                fitScoreTestData.employeeScores,
            );

            // Assert
            expect(result).toMatchObject(
                fitScoreTestData.processedDomainScores,
            );
        });
    });

    describe('getCultureFitEmployeeScore', () => {
        it('should return culture fit score', () => {
            // Act
            const result = FitScoreService.getCultureFitEmployeeScore(
                fitScoreTestData.employeeScores,
                cultureFitRecipeTestData.validCultureFitScoreRecipe,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe('mapParticipant', () => {
        it('should return participant with score', () => {
            // Arrange
            const partialScoreOutput: ParticipantPartialScoreInput = {
                score: fitScoreTestData.score,
                score_dimension: fitScoreTestData.scoreDimension,
                score_outcome: {
                    personality_result:
                        fitScoreTestData.computeCultureFitDomainScores,
                },
            };

            // Act
            const result = FitScoreService.mapParticipant(
                fitScoreTestData.participant,
                partialScoreOutput,
            );

            // Assert
            expect(result).toMatchSnapshot({
                created_at: expect.any(Date),
                updated_at: expect.any(Date),
                assessment_completed_at: expect.any(Date),
            });
        });
    });

    describe('validateWorkStyle', () => {
        it('should return true if valid employee scores is given', () => {
            // Act
            const isWorkStyleValid = FitScoreService.validateWorkStyle(
                employeeScoreTestData.employeeScores,
            );

            // Assert
            expect(isWorkStyleValid).toBeTrue();
        });

        it('should return false if empty employee scores is given', () => {
            // Act
            const isWorkStyleValid = FitScoreService.validateWorkStyle([]);

            // Assert
            expect(isWorkStyleValid).toBeFalse();
        });
    });

    describe('validateWorkValue', () => {
        it('should return true if valid employee scores is given', () => {
            // Act
            const isWorkValueValid = FitScoreService.validateWorkValue(
                employeeScoreTestData.employeeScores,
            );

            // Assert
            expect(isWorkValueValid).toBeTrue();
        });

        it('should return false if invalid employee scores is given', () => {
            // Act
            const isWorkValueValid = FitScoreService.validateWorkValue([]);

            // Assert
            expect(isWorkValueValid).toBeFalse();
        });
    });

    describe('validateWorkInterest', () => {
        it('should return true if valid employee scores is given', () => {
            // Act
            const isWorkInterestValid = FitScoreService.validateWorkInterest(
                employeeScoreTestData.employeeScores,
            );

            // Assert
            expect(isWorkInterestValid).toBeTrue();
        });

        it('should return false if invalid employee scores is given', () => {
            // Act
            const isWorkInterestValid = FitScoreService.validateWorkInterest(
                [],
            );

            // Assert
            expect(isWorkInterestValid).toBeFalse();
        });
    });

    describe('validatePulsifiDefault', () => {
        it('should pass validate role fit recipe', () => {
            // Act
            const shouldPassedValidatePulsifiDefault =
                FitScoreService.validatePulsifiDefault(
                    employeeScoreTestData.employeeScores,
                    employeeScoreTestData.ingredientFrameworkRecipe,
                );

            // Assert
            expect(shouldPassedValidatePulsifiDefault).toBeTrue();
        });

        it('should pass validate culture fit recipe, since there is no ingredient framework as pulsifi_default', () => {
            // Act
            const shouldPassedValidatePulsifiDefault =
                FitScoreService.validatePulsifiDefault(
                    employeeScoreTestData.employeeScores,
                    [],
                );

            // Assert
            expect(shouldPassedValidatePulsifiDefault).toBeTrue();
        });
    });

    describe('validateIfFitScoreIsReadyToBeComputed', () => {
        // Arrange
        const isCultureFit = true;

        it('should pass validate role fit recipe', () => {
            // Arrange
            const newRoleFitRecipe =
                roleFitRecipeTestData.validRoleFitScoreRecipe.recipe.map(
                    (recipe) => ({
                        ...recipe,
                        ingredient_framework:
                            recipe.ingredient_framework ??
                            IngredientFramework.PULSIFI_DEFAULT,
                    }),
                );

            // Act
            const isFitScoreReadyToBeComputed =
                FitScoreService.validateIfFitScoreIsReadyToBeComputed(
                    newRoleFitRecipe,
                    employeeScoreTestData.employeeScores,
                    !isCultureFit,
                );

            // Assert
            expect(isFitScoreReadyToBeComputed).toBeTrue();
        });

        it('should pass validate culture fit recipe', () => {
            // Arrange
            const newCultureFitRecipe =
                cultureFitRecipeTestData.validCultureFitScoreRecipe.recipe.map(
                    (recipe) => ({
                        ...recipe,
                        ingredient_framework:
                            recipe.ingredient_framework ??
                            IngredientFramework.PULSIFI_DEFAULT,
                    }),
                );

            // Act
            const isFitScoreReadyToBeComputed =
                FitScoreService.validateIfFitScoreIsReadyToBeComputed(
                    newCultureFitRecipe,
                    employeeScoreTestData.employeeScores,
                    isCultureFit,
                );

            // Assert
            expect(isFitScoreReadyToBeComputed).toBeTrue();
        });

        it('should failed when employee assessment scores is empty', () => {
            // Act
            const result = () =>
                FitScoreService.validateIfFitScoreIsReadyToBeComputed(
                    roleFitRecipeTestData.validRoleFitScoreRecipe.recipe,
                    [],
                );

            // Assert
            expect(result).not.toBeTrue();
        });

        it('should failed when employee assessment is not completed', () => {
            // Arrange
            const incompleteAssessmentScores = [
                fitScoreTestData.employee1ScoreReasoningNumeric,
                fitScoreTestData.employee1ScoreReasoningLogical,
                fitScoreTestData.employee1ScoreReasoningVerbal,
                fitScoreTestData.employee1ScoreWorkStyle,
            ];

            // Act
            const result = () =>
                FitScoreService.validateIfFitScoreIsReadyToBeComputed(
                    roleFitRecipeTestData.validRoleFitScoreRecipe.recipe,
                    incompleteAssessmentScores,
                );

            // Assert
            expect(result).not.toBeTrue();
        });

        it('should throw error when recipe is empty', () => {
            // Act
            const result = () =>
                FitScoreService.validateIfFitScoreIsReadyToBeComputed(
                    [],
                    employeeScoreTestData.employeeScores,
                );

            // Assert
            expect(result).toThrowErrorMatchingSnapshot();
        });
    });

    describe('shouldCalculateEmployeeFitScore', () => {
        it('should return true when all data is valid', () => {
            // Act
            const shouldSendEvent =
                FitScoreService.shouldCalculateEmployeeFitScore(
                    employeeScoreTestData.program2,
                    cultureFitRecipeTestData.validCultureFitScoreRecipe,
                    employeeScoreTestData.employeeScores,
                    employeeScoreTestData.participant.id,
                );

            // Assert
            expect(shouldSendEvent).toBeTrue();
        });

        it(`should throw error when fit score recipe's recipe is empty`, () => {
            // Arrange
            const emptyRecipe = {
                ...cultureFitRecipeTestData.validCultureFitScoreRecipe,
                recipe: [],
            };

            // Act
            const result = () =>
                FitScoreService.shouldCalculateEmployeeFitScore(
                    employeeScoreTestData.program2,
                    emptyRecipe,
                    employeeScoreTestData.employeeScores,
                    employeeScoreTestData.participant.id,
                );

            // Assert
            expect(result).toThrowErrorMatchingSnapshot();
        });
    });
});
