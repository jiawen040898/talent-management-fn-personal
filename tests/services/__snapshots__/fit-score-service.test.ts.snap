// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FitScoreService getCultureFitEmployeeScore should return culture fit score 1`] = `
{
  "score": 7.0170871,
  "score_dimension": 1,
  "score_outcome": {
    "framework_alias": "polaris",
    "personality_result": [
      {
        "domain_alias": "people_centric",
        "domain_score": 0.7172222,
        "domain_weightage": 0.14285,
        "traits": [
          {
            "trait_alias": "social",
            "trait_framework": "work_interest",
            "trait_score": 0.8125,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "support",
            "trait_framework": "work_value",
            "trait_score": 0.5861111111111111,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "cooperation",
            "trait_framework": "work_style",
            "trait_score": 0.75,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "concern_for_others",
            "trait_framework": "work_style",
            "trait_score": 0.625,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "social_orientation",
            "trait_framework": "work_style",
            "trait_score": 0.8125,
            "trait_weightage": 0.2,
          },
        ],
      },
      {
        "domain_alias": "ownership",
        "domain_score": 0.6222937,
        "domain_weightage": 0.14285,
        "traits": [
          {
            "trait_alias": "investigative",
            "trait_framework": "work_interest",
            "trait_score": 0.3125,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "independence",
            "trait_framework": "work_value",
            "trait_score": 0.6956349206349207,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "relationships",
            "trait_framework": "work_value",
            "trait_score": 0.545,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "leadership",
            "trait_framework": "work_style",
            "trait_score": 0.7458333333333333,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "integrity",
            "trait_framework": "work_style",
            "trait_score": 0.8125,
            "trait_weightage": 0.2,
          },
        ],
      },
      {
        "domain_alias": "learning",
        "domain_score": 0.7294396000000001,
        "domain_weightage": 0.14285,
        "traits": [
          {
            "trait_alias": "working_conditions",
            "trait_framework": "work_value",
            "trait_score": 0.6603174603174603,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "achievement_effort",
            "trait_framework": "work_style",
            "trait_score": 0.8229166666666666,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "initiative",
            "trait_framework": "work_style",
            "trait_score": 0.7678571428571429,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "independence",
            "trait_framework": "work_style",
            "trait_score": 0.6666666666666666,
            "trait_weightage": 0.25,
          },
        ],
      },
      {
        "domain_alias": "avant_garde",
        "domain_score": 0.6556796,
        "domain_weightage": 0.14285,
        "traits": [
          {
            "trait_alias": "realistic",
            "trait_framework": "work_interest",
            "trait_score": 0.3125,
            "trait_weightage": 0.166666667,
          },
          {
            "trait_alias": "independence",
            "trait_framework": "work_value",
            "trait_score": 0.6956349206349207,
            "trait_weightage": 0.166666667,
          },
          {
            "trait_alias": "working_conditions",
            "trait_framework": "work_value",
            "trait_score": 0.6603174603174603,
            "trait_weightage": 0.166666667,
          },
          {
            "trait_alias": "adaptability_flexibility",
            "trait_framework": "work_style",
            "trait_score": 0.828125,
            "trait_weightage": 0.166666667,
          },
          {
            "trait_alias": "innovation",
            "trait_framework": "work_style",
            "trait_score": 0.8125,
            "trait_weightage": 0.166666667,
          },
          {
            "trait_alias": "analytical_thinking",
            "trait_framework": "work_style",
            "trait_score": 0.625,
            "trait_weightage": 0.166666667,
          },
        ],
      },
      {
        "domain_alias": "results_driven",
        "domain_score": 0.7612103,
        "domain_weightage": 0.14285,
        "traits": [
          {
            "trait_alias": "achievement",
            "trait_framework": "work_value",
            "trait_score": 0.8373015873015873,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "recognition",
            "trait_framework": "work_value",
            "trait_score": 0.7083333333333333,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "achievement_effort",
            "trait_framework": "work_style",
            "trait_score": 0.8229166666666666,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "persistence",
            "trait_framework": "work_style",
            "trait_score": 0.8125,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "dependability",
            "trait_framework": "work_style",
            "trait_score": 0.625,
            "trait_weightage": 0.2,
          },
        ],
      },
      {
        "domain_alias": "in_sync",
        "domain_score": 0.6944444999999999,
        "domain_weightage": 0.14285,
        "traits": [
          {
            "trait_alias": "support",
            "trait_framework": "work_value",
            "trait_score": 0.5861111111111111,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "concern_for_others",
            "trait_framework": "work_style",
            "trait_score": 0.625,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "self_control",
            "trait_framework": "work_style",
            "trait_score": 0.8375,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "stress_tolerance",
            "trait_framework": "work_style",
            "trait_score": 0.7291666666666666,
            "trait_weightage": 0.25,
          },
        ],
      },
      {
        "domain_alias": "stakeholder_savvy",
        "domain_score": 0.7319167,
        "domain_weightage": 0.14285,
        "traits": [
          {
            "trait_alias": "enterprising",
            "trait_framework": "work_interest",
            "trait_score": 0.71875,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "relationships",
            "trait_framework": "work_value",
            "trait_score": 0.545,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "leadership",
            "trait_framework": "work_style",
            "trait_score": 0.7458333333333333,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "social_orientation",
            "trait_framework": "work_style",
            "trait_score": 0.8125,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "self_control",
            "trait_framework": "work_style",
            "trait_score": 0.8375,
            "trait_weightage": 0.2,
          },
        ],
      },
    ],
  },
  "score_type": "culture_fit",
}
`;

exports[`FitScoreService mapParticipant should return participant with score 1`] = `
{
  "assessment_completed_at": Any<Date>,
  "company_id": 5,
  "created_at": Any<Date>,
  "created_by": 1,
  "employee_id": "00000000-0000-0000-0000-000000000004",
  "framework_dimension": 1,
  "framework_outcome": [
    {
      "domain_alias": "people_centric",
      "domain_score": 0.7172222,
      "domain_weightage": 0.14285,
      "traits": [
        {
          "trait_alias": "social",
          "trait_framework": "work_interest",
          "trait_score": 0.8125,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "support",
          "trait_framework": "work_value",
          "trait_score": 0.5861111111111111,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "cooperation",
          "trait_framework": "work_style",
          "trait_score": 0.75,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "concern_for_others",
          "trait_framework": "work_style",
          "trait_score": 0.625,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "social_orientation",
          "trait_framework": "work_style",
          "trait_score": 0.8125,
          "trait_weightage": 0.2,
        },
      ],
    },
    {
      "domain_alias": "ownership",
      "domain_score": 0.6222937,
      "domain_weightage": 0.14285,
      "traits": [
        {
          "trait_alias": "investigative",
          "trait_framework": "work_interest",
          "trait_score": 0.3125,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "independence",
          "trait_framework": "work_value",
          "trait_score": 0.6956349206349207,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "relationships",
          "trait_framework": "work_value",
          "trait_score": 0.545,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "leadership",
          "trait_framework": "work_style",
          "trait_score": 0.7458333333333333,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "integrity",
          "trait_framework": "work_style",
          "trait_score": 0.8125,
          "trait_weightage": 0.2,
        },
      ],
    },
    {
      "domain_alias": "learning",
      "domain_score": 0.7294396000000001,
      "domain_weightage": 0.14285,
      "traits": [
        {
          "trait_alias": "working_conditions",
          "trait_framework": "work_value",
          "trait_score": 0.6603174603174603,
          "trait_weightage": 0.25,
        },
        {
          "trait_alias": "achievement_effort",
          "trait_framework": "work_style",
          "trait_score": 0.8229166666666666,
          "trait_weightage": 0.25,
        },
        {
          "trait_alias": "initiative",
          "trait_framework": "work_style",
          "trait_score": 0.7678571428571429,
          "trait_weightage": 0.25,
        },
        {
          "trait_alias": "independence",
          "trait_framework": "work_style",
          "trait_score": 0.6666666666666666,
          "trait_weightage": 0.25,
        },
      ],
    },
    {
      "domain_alias": "avant_garde",
      "domain_score": 0.6556796,
      "domain_weightage": 0.14285,
      "traits": [
        {
          "trait_alias": "realistic",
          "trait_framework": "work_interest",
          "trait_score": 0.3125,
          "trait_weightage": 0.166666667,
        },
        {
          "trait_alias": "independence",
          "trait_framework": "work_value",
          "trait_score": 0.6956349206349207,
          "trait_weightage": 0.166666667,
        },
        {
          "trait_alias": "working_conditions",
          "trait_framework": "work_value",
          "trait_score": 0.6603174603174603,
          "trait_weightage": 0.166666667,
        },
        {
          "trait_alias": "adaptability_flexibility",
          "trait_framework": "work_style",
          "trait_score": 0.828125,
          "trait_weightage": 0.166666667,
        },
        {
          "trait_alias": "innovation",
          "trait_framework": "work_style",
          "trait_score": 0.8125,
          "trait_weightage": 0.166666667,
        },
        {
          "trait_alias": "analytical_thinking",
          "trait_framework": "work_style",
          "trait_score": 0.625,
          "trait_weightage": 0.166666667,
        },
      ],
    },
    {
      "domain_alias": "results_driven",
      "domain_score": 0.7612103,
      "domain_weightage": 0.14285,
      "traits": [
        {
          "trait_alias": "achievement",
          "trait_framework": "work_value",
          "trait_score": 0.8373015873015873,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "recognition",
          "trait_framework": "work_value",
          "trait_score": 0.7083333333333333,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "achievement_effort",
          "trait_framework": "work_style",
          "trait_score": 0.8229166666666666,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "persistence",
          "trait_framework": "work_style",
          "trait_score": 0.8125,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "dependability",
          "trait_framework": "work_style",
          "trait_score": 0.625,
          "trait_weightage": 0.2,
        },
      ],
    },
    {
      "domain_alias": "in_sync",
      "domain_score": 0.6944444999999999,
      "domain_weightage": 0.14285,
      "traits": [
        {
          "trait_alias": "support",
          "trait_framework": "work_value",
          "trait_score": 0.5861111111111111,
          "trait_weightage": 0.25,
        },
        {
          "trait_alias": "concern_for_others",
          "trait_framework": "work_style",
          "trait_score": 0.625,
          "trait_weightage": 0.25,
        },
        {
          "trait_alias": "self_control",
          "trait_framework": "work_style",
          "trait_score": 0.8375,
          "trait_weightage": 0.25,
        },
        {
          "trait_alias": "stress_tolerance",
          "trait_framework": "work_style",
          "trait_score": 0.7291666666666666,
          "trait_weightage": 0.25,
        },
      ],
    },
    {
      "domain_alias": "stakeholder_savvy",
      "domain_score": 0.7319167,
      "domain_weightage": 0.14285,
      "traits": [
        {
          "trait_alias": "enterprising",
          "trait_framework": "work_interest",
          "trait_score": 0.71875,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "relationships",
          "trait_framework": "work_value",
          "trait_score": 0.545,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "leadership",
          "trait_framework": "work_style",
          "trait_score": 0.7458333333333333,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "social_orientation",
          "trait_framework": "work_style",
          "trait_score": 0.8125,
          "trait_weightage": 0.2,
        },
        {
          "trait_alias": "self_control",
          "trait_framework": "work_style",
          "trait_score": 0.8375,
          "trait_weightage": 0.2,
        },
      ],
    },
  ],
  "framework_score": 0.1,
  "id": "00000000-0000-0000-0000-000000000004",
  "program_id": 4,
  "updated_at": Any<Date>,
  "updated_by": 1,
}
`;

exports[`FitScoreService shouldCalculateEmployeeFitScore should throw error when fit score recipe's recipe is empty 1`] = `"Missing recipes for employee fit score computation"`;

exports[`FitScoreService validateIfFitScoreIsReadyToBeComputed should throw error when recipe is empty 1`] = `"Missing recipes for employee fit score computation"`;
